src/core/utils/checks/charge_neutrality.hpp

```diff
@@ -1,33 +1,33 @@
 #include <cmath>
-
-#include "PartCfg.hpp"
+#include <limits>
 
 namespace Utils {
 
-inline void check_charge_neutrality(PartCfg &partCfg) {
-  double total_charge = 0.0;
-  double min_charge = 0.0;
-  for (auto const &particle : partCfg) {
-    total_charge += particle.p.q;
-    double abs_charge = std::abs(particle.p.q);
-    if (min_charge == 0.0 and abs_charge != 0.0)
-      min_charge = abs_charge;
-    if (abs_charge < min_charge and abs_charge != 0.0)
-      min_charge = particle.p.q;
+template<typename ParticleRange, typename Particle>
+inline void check_charge_neutrality(ParticleRange &prange) {
```

Comment:
Better use
```cpp
template<typename ParticleRange>
inline void check_charge_neutrality(ParticleRange &prange) {
    using Particle = typename ParticleRange::value_type;
```
then you don't have to specify explicit template arguments.

Reply:
Just out of curiosity, why would you put in the stopper?

Reply:
To prevent people from trying to interfere with template argument deduction.  Way too often you see things like `std::make_pair<int,std::string>(1,"Hello");`.